fs/shm/shmfs.h

```diff
@@ -0,0 +1,72 @@
+/****************************************************************************
+ * fs/shm/shmfs.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __FS_SHM_SHMFS_H
+#define __FS_SHM_SHMFS_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdlib.h>
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+extern const struct file_operations shmfs_operations;
+
+/****************************************************************************
+ * Public Types
+ ****************************************************************************/
+
+struct shmfs_object_s
```

Comment:
shm_object_s

Reply:
I think this object is just private data type for shmfs, so kept it as shmfs_object. To extend it to be something for wider scope would require also specifying the interfaces for managing these objects differently...

Reply:
ok, let's unify the prefix to shmfs_ except shm_open/shm_unlink